### PR TITLE
Remove deprecated and backwards compatibility code.

### PR DESCRIPTION
**Pull request by ADmad**
Prepare for RC baby!

### CODE REVIEW

**Comment by antograssiot** [comment]
nice cleanup :+: 
you have a small cs error

**Comment by lorenzo** [comment]
yay!

File: tests/TestCase/Error/DebuggerTest.php [hunk 1]
```diff
@@ -245,66 +147,6 @@ public function testOutputAsException()
...
-
-        $expected = [
-            '<error',
-            '<code', '8', '/code',
-            '<file', 'preg:/[^<]+/', '/file',
-            '<line', '' . ((int)__LINE__ - 7), '/line',
-            'preg:/Undefined variable:\s+foo/',
-            '/error'
-        ];
-        $this->assertHtml($expected, $result, true);
```
Comment (markstory) [inline tests/TestCase/Error/DebuggerTest.php]: Doesn't removing these test leave addFormat() mostly untested?

Comment (lorenzo) [inline tests/TestCase/Error/DebuggerTest.php]: How can it be tested now?

Comment (markstory) [inline tests/TestCase/Error/DebuggerTest.php]: By calling outputError(). I can re add tests later today.

Comment (lorenzo) [inline tests/TestCase/Error/DebuggerTest.php]: @markstory ok, cool. I will start writing the release announcement

Comment (hdogan) [inline tests/TestCase/Error/DebuggerTest.php]: > I will start writing the release announcement

:dancers: :metal:  :clap: :tada:

Comment (markstory) [inline tests/TestCase/Error/DebuggerTest.php]: @lorenzo Sounds good.

Comment (ADmad) [inline tests/TestCase/Error/DebuggerTest.php]: @markstory Sorry about the removed tests, I got a bit lazy in investigating and updating them.

Comment (markstory) [inline tests/TestCase/Error/DebuggerTest.php]: No worries I spent some time on removing the deprecated code last night as well, but got too tire before I finished. However, I did get these tests working, so we can still use that part of it.

**Pull request merged**